packages/headless/src/features/insight-user-actions/insight-user-actions-actions.ts

```diff
@@ -0,0 +1,69 @@
+import {ArrayValue, NumberValue} from '@coveo/bueno';
+import {createAction, createAsyncThunk} from '@reduxjs/toolkit';
+import {isErrorResponse} from '../../api/search/search-api-client';
+import {AsyncThunkInsightOptions} from '../../api/service/insight/insight-api-client';
+import {InsightUserActionsResponse} from '../../api/service/insight/user-actions/user-actions-response';
+import {
+  ConfigurationSection,
+  InsightConfigurationSection,
+  InsightUserActionSection,
+} from '../../state/state-sections';
+import {nonEmptyString, validatePayload} from '../../utils/validate-payload';
+import {buildFetchUserActionsRequest} from './insight-user-actions-request';
+
+const numberValue = new NumberValue({required: true, min: 0});
+
+interface RegisterUserActionsPayload {
+  ticketCreationDate: string;
+  numberSessionsBefore?: number;
+  numberSessionsAfter?: number;
+  excludedCustomActions?: string[];
+}
+
+const registerUserActionsPayloadSchema = {
+  ticketCreationDate: nonEmptyString,
```

Comment:
We should validate the date format as well.

Reply:
I added an option to validate ISO datetime strings in Bueno. [See here](https://github.com/coveo/ui-kit/pull/2637/commits/0fa788fdb05e8c22d2199c5eadfd71473efb6bcf#diff-77f7c6dbefaa1941ae6e15438b7aa4c0dc17d9da6e387f5b39f0aabc6b99d36c). This way we could use it in quantic as well to validate the component prop. What do you think?

Reply:
I didn't review all the code, but the validator part looks good.